src/Homework/Week3/Pb2/Customer.java

```diff
@@ -0,0 +1,58 @@
+package Homework.Week3.Pb2;
+
+import java.util.List;
+
+public class Customer {
+    private String name;
+    private MemberShip myMemberShip;
+    {
+        myMemberShip = MemberShip.NONE;
+    }
+    private Basket myBasket;
+
+    Customer(String name, MemberShip memberShip)
+    {
+        myMemberShip = memberShip;
+        myBasket = new Basket();
+    }
+
+    void addToBasket(ItemForSale item) {
+        if(item == null) {
+            System.out.println("The item added to the basket is null");
+            return;
+        }
+        myBasket.addItem(item);
+    }
+
+    float basketPrice()
+    {
+        if(myMemberShip.equals(MemberShip.NONE))
```

Comment:
1. daca myMemberShip e null, un NullPointerException e aruncat: poti evita asta daca inversezi comparatia:
`MemberShip.NONE.equals(myMemberShip)`
2. tinand cont ca lucreziz cu enum-uri, poti folosi in cazul asta pentru comparatie == in loc de equals si astfel ai evita cazul de mai sus, cu NullPointerException